src/compiler/checker.ts

```diff
@@ -11175,7 +11175,7 @@ namespace ts {
             // Abstract methods cannot have an implementation.
             // Extra checks are to avoid reporting multiple errors relating to the "abstractness" of the node.
             if (node.flags & NodeFlags.Abstract && node.body) {
-                error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
+                error(node.body, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));

```

Comment:
that is not much better than just node, the body could be a whole file. with the other change, the new range will be just the name, which is a better fit.